File: frontend/erpclink-web/src/app/features/specialties/models/specialty.models.ts
export interface CreateSpecialtyRequest {
  code: string;
  name: string;
  description?: string | null;
}

export interface UpdateSpecialtyRequest {
  name: string;
  description?: string | null;
}

export interface SpecialtyDto {
  id: string;
  organizationId: string;
  code: string;
  name: string;
  description: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

File: src/Modules/Doctors/ErpClink.Modules.Doctors.Application/Specialties/Models/SpecialtyModels.cs
namespace ErpClink.Modules.Doctors.Application.Specialties.Models;

public sealed record CreateSpecialtyRequest(string Code, string Name, string? Description);
public sealed record UpdateSpecialtyRequest(string Name, string? Description);

public sealed record SpecialtyDto(
    Guid Id,
    Guid OrganizationId,
    string Code,
    string Name,
    string? Description,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);
